### PR TITLE
OPAL-530

### PR DESCRIPTION
**Pull request by AM-USAF**
created pytest to catch bug and updated track_basket to fix bug

### CODE REVIEW

**Review by hh-usaf — APPROVED** [review]

**Review by JA-USAF — APPROVED** [review]
😎

File: weave/tests/test_index_pandas.py [hunk 1]
```diff
@@ -221,3 +222,39 @@ def test_is_index_current(test_pantry):
...
+    )
+    # Upload basket with new index
+    with tempfile.TemporaryDirectory(dir=".") as tmpdir:
+        tmp_file_path = os.path.join(tmpdir, "temp_basket.txt")
+        with open(tmp_file_path, "w", encoding="utf-8") as tmp_file:
+            pantry2.upload_basket(
+                upload_items=[{"path":tmp_file.name, "stub":False}],
+                basket_type="test-1",
+            )
+    assert len(pantry2.index.to_pandas_df()) == 2
```
Comment (hh-usaf) [inline weave/tests/test_index_pandas.py:260]: When I ran it locally this test didn't pass. It only had the length of one.

Comment (AM-USAF) [reply weave/tests/test_index_pandas.py:260]: It is still passing locally for me, which version of weave are you running? - it should fail for version 1.7.2 and pass for the changes in my branch (1.7.3)

Comment (hh-usaf) [reply weave/tests/test_index_pandas.py:260]: I was on your branch(1.7.3). I'll continue to look at it, since I think I passed when I ran it yesterday.

**Review by hh-usaf — APPROVED** [review]

**Pull request merged**